lib/Runtime/Language/JavascriptConversion.cpp

```diff
@@ -1256,6 +1256,11 @@ namespace Js
         return JavascriptMath::ToInt32Core(T1);
     }
 
+    __int64 JavascriptConversion::ToInt64(double T1)
+    {
+        return JavascriptMath::TryToInt64(T1);
```

Comment:
What should we do if `NumberUtilities::IsValidTryToInt64` is false for the result of this call?

Reply:
Seems like TryToInt64 would handle this case? In the case of overflows/Nans, it would just return canonical values? We should however document it

Reply:
`TryToInt64` doesn't do anything special for x64 builds, it's simply a cast from `double` to `int64_t`.  There are special paths for x86 (which can use SSE3) and ARM (which has different behavior), but in the base case it's just a cast.

For `ToInt32` it ends up normalizing all of these cases down to zero without returning an error, but I'm not sure if that's spec'd anywhere.